app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -77,26 +76,24 @@ class SearchRepositoriesViewModel(
             )
             .onStart { emit(UiAction.Scroll(currentQuery = lastQueryScrolled)) }
 
-        state = searches
-            .flatMapLatest { search ->
-                combine(
-                    queriesScrolled,
-                    searchRepo(queryString = search.query),
-                    ::Pair
-                )
-                    // Each unique PagingData should be submitted once, take the latest from
-                    // queriesScrolled
-                    .distinctUntilChangedBy { it.second }
-                    .map { (scroll, pagingData) ->
-                        UiState(
-                            query = search.query,
-                            pagingData = pagingData,
-                            lastQueryScrolled = scroll.currentQuery,
-                            // If the search query matches the scroll query, the user has scrolled
-                            hasNotScrolledForCurrentSearch = search.query != scroll.currentQuery
-                        )
-                    }
-            }
+        val pagingDataFlow = searches
+            .flatMapLatest { searchRepo(queryString = it.query) }
+            .cachedIn(viewModelScope)
+
+        state = combine(
+            searches,
+            pagingDataFlow,
+            queriesScrolled,
+            ::Triple
+        ).map { (search, pagingData, scroll) ->
+            UiState(
+                query = search.query,
+                pagingData = pagingData,
```

Comment:
i really don't like we are putting a Flow into a UIState. UIState is meant to be state and that is not possible w/ paging w/o additional APIs. But then some external dev will take it as an example and keep putting flows into state objects.
It is better to separate this until it is possible.